### ISSUE TITLE
Can libraries be loaded from sub-directories in "/lib" etc.?

### ISSUE THREAD

**Issue by samwalls** [MightyPirates/OpenComputers#2022]
Well, to be clear, I know they can't by default because I've tried it, i.e.:

Say this is a library I have in the location "/lib/test/try.lua"

```
local try = {}

local function try.please()
  print("yes?")
end

return try
```

and say I want to invoke it:

```
local t = require("test/try")
t.please()
```

Running this (of course, after rebooting as well) gives back the error
http://imgur.com/a/sHXZO

I don't know if this is the intended behaviour or not, because it works in standard lua... Is there a way I can get the OS to load libraries that are namespaced like this?

**Comment by payonel** [comment]
we use standard lua :)

and yes it can work, and will work, and in fact is working, and telling you what went wrong

`/home/test:1: module 'test/try' not found:`
This is printed because require aborted when it failed to successfully load module 'test/try'

Besides checking the package path, package require checks preload and delayed - the error is also telling you that it couldn't find the module in these areas either

Following the faiilure to find the module, `require` will also report the issues it encountered during its effort to load any module that matched 'test/try'

`/lib/test/try.lua:3: '(' expected near '.'`

This is the compilation error encountered when `require` tried to load `/lib/test/try.lua` on line 3

In other words, your test code is invalid lua code. Why? you don't define table methods using local

`function try.please() end`

not

`local function try.please() end`

**Comment by samwalls** [comment]
No matter how much I consider myself getting better at programming, lua always finds a way of leaving me stumped... Thank you anyway!
